webpush/encrypt.go

```diff
@@ -144,16 +170,25 @@ type EncryptionResult struct {
 //    - https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding
 //    - https://en.wikipedia.org/wiki/Elliptic_curve_Diffie%E2%80%93Hellman
 //    - https://tools.ietf.org/html/draft-ietf-webpush-encryption
-func Encrypt(sub *Subscription, message string) (*EncryptionResult, error) {
+func Encrypt(sub *Subscription, message string, encoding ContentEncoding) (*EncryptionResult, error) {
 	plaintext := []byte(message)
+
+	var maxPayloadLength int
```

Comment:
There's no reason to define a var with initial zero value here. A more Go idiomatic style would be the following:

```go
maxlen = aesgcmMaxPayloadLength
if encoding == AES128GCM {
  maxlen = aes128gcmMaxPayloadLength
}
```